Assets/Scripts/Models/EventAction.cs

```diff
@@ -123,15 +123,25 @@ public void Deregister(string actionName, string luaFunc)
     /// <param name="actionName">Name of the action being triggered.</param>
     /// <param name="target">Object, passed to LUA function as 1-argument (TODO: make it an object).</param>
     /// <param name="deltaTime">Time since last Trigger of this event.</param>
-    public void Trigger(string actionName, Furniture target, float deltaTime = 0f)
+    public void Trigger<T>(string actionName, T target, float deltaTime = 0f)
```

Comment:
It might, but this works at least as well. When the T is one of the arguments passed, you don't even need to include the brackets (so it would still be called as `Trigger("OnUpdate",need)`).